comp/core/hostname/component.go

```diff
@@ -6,20 +6,11 @@
 // Package hostname exposes hostname.Get() as a component.
 package hostname
 
-import (
-	"context"
-
-	"github.com/DataDog/datadog-agent/pkg/util/hostname"
-)
+import "github.com/DataDog/datadog-agent/comp/core/hostname/hostnameinterface"
 
 // team: agent-shared-components
 
 // Component is the component type.
 type Component interface {
```

Comment:
Note that this `Component` interface is in a package that doesn't depend on fx.
If you want to be able to create this component without using fx you can just export [newHostnameService()](https://github.com/DataDog/datadog-agent/blob/7.50.0/comp/core/hostname/hostnameimpl/service.go#L47).

Do you have to not depend on fx package (If yes there is also a solution)?

Reply:
There's no limitation on fx for our use case, but we can't use newHostnameService() in `pkg/logs` because `hostnameimpl` is not a module and we're trying to convert packages in `pkg/logs` into modules.